### PR TITLE
Memory allocation improvement for `decode_libsvm`

### PR DESCRIPTION
**Pull request by yongtang**
This fix is an improvement to #14330. Previously, string split was handled through `str_util::Split`, which may incur unnecessary memory allocations. This fix uses StringPiece instead.

See comment https://github.com/tensorflow/tensorflow/pull/14330#pullrequestreview-79877956 for reference.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow/contrib/libsvm/kernels/decode_libsvm_op.cc [hunk 1]
```diff
@@ -46,34 +46,44 @@ class DecodeLibsvmOp : public OpKernel {
...
     std::vector<std::pair<int64, int64>> out_indices;
     for (int i = 0; i < input_flat.size(); ++i) {
-      std::vector<string> entries =
-          str_util::Split(input_flat(i), " ", str_util::SkipEmpty());
-      OP_REQUIRES(ctx, !entries.empty(),
-                  errors::InvalidArgument("No entries found for input[", i,
+      StringPiece line(input_flat(i));
+      str_util::RemoveWhitespaceContext(&line);
+      StringPiece piece;
+      ;
```
Comment (mrry) [inline tensorflow/contrib/libsvm/kernels/decode_libsvm_op.cc]: Can you delete this empty statement?

**Review by mrry — CHANGES_REQUESTED** [review]
Thanks for the improvement! Just one minor nit.

File: tensorflow/contrib/libsvm/kernels/decode_libsvm_op.cc (hunk 1 above)
Comment (yongtang) [reply tensorflow/contrib/libsvm/kernels/decode_libsvm_op.cc]: Thanks @mrry! Updated.

**Review by mrry — APPROVED** [review]
Thanks! Looks good to me!

**Comment by caisq** [comment]
Thanks for the review, @mrry

**Pull request merged**